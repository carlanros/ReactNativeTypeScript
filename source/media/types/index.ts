// media/types/index.ts

export enum MediaConstants {
  /**
   * Login failed or was cancelled
   */
  MEDIA_FAILED = 'MEDIA_FAILED',
  /**
   * Login state: authentication succeeded
   *
   * @static
   * @memberof MediaConstants
   */
  MEDIA_SUCCESS = 'MEDIA_SUCCESS',

  /**
   * Login state: authentication in progress
   *
   * @static
   * @memberof MediaConstants
   */
  MEDIA_BUSY = 'Kollar vem du Ã¤r',
}

export interface IMediaState {
  cancelled: boolean;
  height: number;
  width: number;
  uri: string;

}
export interface IMediaAction extends IMediaState {
  type: MediaConstants;
}
