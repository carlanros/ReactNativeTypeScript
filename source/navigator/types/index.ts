export interface INavigationState {
  /**
   * Current scene object in navigation
   *
   * @type {object}
   * @memberof INavigationReducer
   */
  scene: object;
}

export interface INavigationAction extends INavigationState {
  type: string;
}
