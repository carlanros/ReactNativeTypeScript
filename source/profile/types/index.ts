// profile/types/index
import * as types from '../../Types';

export interface IProfileProps extends types.IProps {
  Logout?: () => any;
}
