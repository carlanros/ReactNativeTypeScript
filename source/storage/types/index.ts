// storage/types/index

/**
 * App storage Redux action types
 * @type {StorageConstants}
 * @public
 * @type {Object}
 */
export enum StorageConstants {
    STORAGE_GOTKEY = 'STORAGE = GOT KEY',
    STORAGE_REMOVEDKEY = 'STORAGE = REMOVED KEY',
    STORAGE_REMOVED_FAILED = 'STORAGE = REMOVED KEY FAILED',
    STORAGE_NOSUCHKEY = 'STORAGE = NO SUCH KEY',
    STORAGE_SAVE_FAILED = 'STORAGE = SAVE FAILED',
    STORAGE_SAVED = 'STORAGE = SAVED KEY',
    STORAGE_USER_KEY = 'STORAGE KEY = USER',
    STORAGE_BUSY = 'Friskar upp minnet',
  }

export interface IKey {
/**
 * The key in app storage
 *
 * @type {string}
 */
  key: string;
}

/**
 * Reducer properties passed down from StorageReducer
 * @typedef IStorageReducer
 * @type {Object}
 * @public
 * @export
 * @property {Object} value - the app storage value
 * @property {string} key - The app storage key
 */
export interface IStorageState extends IKey {

  /**
   * Value to save / load from app storage
   *
   * @type {object?}
   */
  value?: object;
}

export interface IStorageAction extends IStorageState {
  type: StorageConstants;
}