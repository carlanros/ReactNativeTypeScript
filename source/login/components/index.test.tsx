// login/components/index
import React from 'react';
import { TextInput } from 'react-native';
import { shallow } from 'enzyme';

import * as types from '../../types';
import { Login } from './';

jest.unmock('react-native');

function setup() {
  const props: types.IProps = {
    login: { isLoggedIn: false },
    utility: { isBusy: false, hasError: false, errorMessage: null },
  };

  const enzymeWrapper = shallow(<Login {...props} />);

  return {
    props,
    enzymeWrapper,
  };
}

describe('components', () => {
  /**
   * Test component rendering. Properties of children might be tested by importing their type:
   *
   *    const busyProps = enzymeWrapper.find(BusyIndicator).props();
   *    expect(busyProps.isBusy).toBe(false);
   */
  describe('Login', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
