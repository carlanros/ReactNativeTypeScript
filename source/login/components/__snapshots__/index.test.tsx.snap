// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Login should render self and subcomponents 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    login={
      Object {
        "isLoggedIn": false,
      }
    }
    utility={
      Object {
        "errorMessage": null,
        "hasError": false,
        "isBusy": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "padding": 10,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "flex-start",
                },
                Object {
                  "padding": 10,
                },
              ]
            }
          >
            Login
          </Text>
          <UNDEFINED
            button={true}
            onPress={[Function]}
            style={
              Object {
                "width": 200,
              }
            }
            title="Sign In With Facebook"
            type="facebook"
          />
        </View>,
        <busyIndicator
          isBusy={false}
          message={undefined}
        />,
      ],
      "hideClose": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "flex-start",
                  },
                  Object {
                    "padding": 10,
                  },
                ]
              }
            >
              Login
            </Text>,
            <UNDEFINED
              button={true}
              onPress={[Function]}
              style={
                Object {
                  "width": 200,
                }
              }
              title="Sign In With Facebook"
              type="facebook"
            />,
          ],
          "style": Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "padding": 10,
            },
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Login",
              "ellipsizeMode": "tail",
              "style": Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "flex-start",
                },
                Object {
                  "padding": 10,
                },
              ],
            },
            "ref": null,
            "rendered": "Login",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "button": true,
              "onPress": [Function],
              "style": Object {
                "width": 200,
              },
              "title": "Sign In With Facebook",
              "type": "facebook",
            },
            "ref": null,
            "rendered": null,
            "type": undefined,
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "isBusy": false,
          "message": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "padding": 10,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "flex-start",
                  },
                  Object {
                    "padding": 10,
                  },
                ]
              }
            >
              Login
            </Text>
            <UNDEFINED
              button={true}
              onPress={[Function]}
              style={
                Object {
                  "width": 200,
                }
              }
              title="Sign In With Facebook"
              type="facebook"
            />
          </View>,
          <busyIndicator
            isBusy={false}
            message={undefined}
          />,
        ],
        "hideClose": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "flex-start",
                    },
                    Object {
                      "padding": 10,
                    },
                  ]
                }
              >
                Login
              </Text>,
              <UNDEFINED
                button={true}
                onPress={[Function]}
                style={
                  Object {
                    "width": 200,
                  }
                }
                title="Sign In With Facebook"
                type="facebook"
              />,
            ],
            "style": Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "padding": 10,
              },
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Login",
                "ellipsizeMode": "tail",
                "style": Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "flex-start",
                  },
                  Object {
                    "padding": 10,
                  },
                ],
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "button": true,
                "onPress": [Function],
                "style": Object {
                  "width": 200,
                },
                "title": "Sign In With Facebook",
                "type": "facebook",
              },
              "ref": null,
              "rendered": null,
              "type": undefined,
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isBusy": false,
            "message": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
